driver-core/src/main/com/mongodb/internal/connection/ProtocolHelper.java

```diff
@@ -264,7 +265,9 @@ public static MongoException createSpecialException(final BsonDocument response,
         } else if (isNotPrimaryError(errorCode, errorMessage)) {
             return new MongoNotPrimaryException(response, serverAddress);
         } else if (response.containsKey("writeConcernError")) {
-            return createSpecialException(response.getDocument("writeConcernError"), serverAddress, "errmsg");
+            return createSpecialException(response.getDocument("writeConcernError").clone()
```

Comment:
It's pretty hacky to clone the writeConcernError document so we can append errorLabels to it, but it's the most localized fix.  Happy to entertain other ideas though